### PR TITLE
Creating and updating lambda READMEs

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

Just updating / initializing README for the lambdas.

Not sure if other modules like `text-to-code`, `augmentation`, and `validation`, etc. Happy to make a ticket for that work. I actually don't know if we've chatted about if/how we want readable docs for the lambdas given that we aren't currently pursuing API stuff. Figured the lambdas at least will still be helpful to have more granular.

Anyway, maybe food for thought for an eng sync and/or parking lot.

## Related Issues

Closes #[Link any related issues or tasks from your project management system.]

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: packages/augmentation-lambda/README.md [hunk 1]
```diff
@@ -1,19 +1,62 @@
...
+
+The Lambda reads these environment variables:
+
+```text
+S3_BUCKET
+TTC_INPUT_PREFIX
+TTC_OUTPUT_PREFIX
+AUGMENTED_EICR_PREFIX
+AUGMENTATION_METADATA_PREFIX
+```
```
Comment (nickclyde) [inline packages/augmentation-lambda/README.md:53]: I think this env var list is missing `AWS_REGION`

File: packages/augmentation-lambda/README.md [hunk 2]
```diff
@@ -1,19 +1,62 @@
...
+
+This package is responsible for:
+
+- Reading the SQS-wrapped S3 event.
+- Extracting the S3 persistence ID from the TTC output object key.
+- Loading TTC output from S3.
+- Loading the original eICR XML from S3.
+- Parsing nonstandard code translations from TTC output.
+- Running eICR augmentation through the core augmentation package.
+- Managing a S3-derived persistence ID for the augmented document and set IDs.
```
Comment (nickclyde) [inline packages/augmentation-lambda/README.md]: I found "and set IDs" a little ambiguous on first read. I believe this is referring to the eICR `setId`/`versionNumber` header elements that `EICRAugmenter` rewrites deterministically from the `persistence_id` seed, but it could also be read as "the augmented document and the set IDs (plural)." Could we phrase it as something like "Seeding deterministic eICR document IDs (id, setId, versionNumber) from the persistence ID" so it's clearer what's being managed?

File: packages/index-lambda/README.md [hunk 3]
```diff
@@ -1,19 +1,116 @@
...
+- [Pre-requisites](#pre-requisites)
+- [Responsibilities](#responsibilities)
+- [Index Behavior](#index-behavior)
+- [Lambda Actions](#lambda-actions)
+- [Environment Variables](#environment-variables)
+- [Tests](#tests)
+
 ## Overview
 
 🚧 Under Construction 🚧
```
Comment (nickclyde) [inline packages/index-lambda/README.md]: Now that this README has substantive content, should we remove the "Under Construction" banner? Same question for the lambda-handler README

File: packages/lambda-handler/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+- `check_s3_object_exists`
+- `get_eventbridge_data_from_s3_event`
+- `get_persistence_id`
+
+`get_persistence_id` extracts the persistence ID from an S3 object key by removing the expected pipeline prefix.
+
+For example:
+
+```text
+TTCInput/2026/01/01/0026b704-f510-4494-8d21-11d27217d96e
```
Comment (nickclyde) [inline packages/lambda-handler/README.md]: Nit: `TTCInput/` is the old prefix we had, the new defaults are `TextToCodeSubmissionV2/`, `ValidationResponseV2/`, `TTCAugmentationMetadataV2/`, etc.

File: packages/lambda-handler/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+## S3 Helpers
+
+The package includes helpers for common S3 operations:
+
+- `create_s3_client`
+- `get_file_content_from_s3`
+- `put_file`
+- `check_s3_object_exists`
+- `get_eventbridge_data_from_s3_event`
+- `get_persistence_id`
```
Comment (nickclyde) [inline packages/lambda-handler/README.md:57]: Nit: `get_eventbridge_data_from_s3_event` and `get_persistence_id` aren't really S3 helpers, they're EventBridge/key-parsing helpers that happen to be used alongside S3. Would it be worth renaming this section to something broader like "S3 and Event Helpers"?

File: packages/text-to-code-lambda/README.md [hunk 6]
```diff
@@ -1,36 +1,149 @@
...
+
+This is required by the source-bucket directed model: the Lambda reads inputs from, and writes outputs back to, whichever bucket the event originated from.
+
+When onboarding a new bucket, update the Lambda's IAM policy to grant read/write access to that bucket.
+
+The Lambda also needs permissions to access the configured OpenSearch cluster.
+
+## Logging
+
+Every TTC invocation logs the record count at the start of processing.
```
Comment (nickclyde) [inline packages/text-to-code-lambda/README.md]: Suggestion for a little more specificity: "Every TTC invocation logs the record count at the start. For each record, the Lambda logs the event bucket, triggering object key, and derived persistence ID as structured fields..."

**Review by nickclyde — COMMENTED** [review]
Awesome, thanks for adding these Rob!!

File: packages/index-lambda/README.md [hunk 7]
```diff
@@ -1,19 +1,116 @@
...
+INDEX_NAME
+OPENSEARCH_ENDPOINT_URL
+```
+
+## Tests
+
+Run the package tests with:
+
+```bash
+uv run pytest packages/index-lambda/tests
```
Comment (JNygaard-Skylight) [inline packages/index-lambda/README.md]: You can do this with the just command with `just all unit packages/index-lambda/tests`
```suggestion
just all unit packages/index-lambda/tests
```

File: packages/lambda-handler/README.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+S3_ENDPOINT_URL
+OPENSEARCH_ENDPOINT_URL
+```
+
+## Tests
+
+Run the package tests with:
+
+```bash
+uv run pytest packages/lambda-handler/tests
```
Comment (JNygaard-Skylight) [inline packages/lambda-handler/README.md]:
```suggestion
just test all packages/lambda-handler/tests
```

File: packages/text-to-code-lambda/README.md [hunk 9]
```diff
@@ -1,36 +1,149 @@
...
+| `S3_ENDPOINT_URL`         | No       | —                            | Optional custom S3 endpoint for local or mocked environments |
+| `OPENSEARCH_ENDPOINT_URL` | Yes      | —                            | OpenSearch cluster endpoint                                  |
+| `OPENSEARCH_INDEX`        | No       | `ttc-index`                  | OpenSearch index name for vector search                      |
+
+## Tests
+
+Run the package tests with:
+
+```bash
+uv run pytest packages/text-to-code-lambda/tests
```
Comment (JNygaard-Skylight) [inline packages/text-to-code-lambda/README.md]:
```suggestion
just test all packages/text-to-code-lambda/tests
```

File: packages/index-lambda/README.md [hunk 10]
```diff
@@ -1,19 +1,116 @@
 # DIBBs Index Initialization
 
-**General disclaimer** This repository was created for use by CDC programs to collaborate on public health related projects in support of the [CDC mission](https://www.cdc.gov/about/cdc/#cdc_about_cio_mission-our-mission). GitHub is not hosted by the CDC, but is a third party website used by CDC and its partners to share information and collaborate on software. CDC use of GitHub does not imply an endorsement of any one particular service, product, or enterprise.
```
Comment (JNygaard-Skylight) [inline packages/index-lambda/README.md:3]: Why remove the Markdown link?

File: packages/augmentation-lambda/README.md [hunk 11]
```diff
@@ -1,19 +1,62 @@
-# DIBBs Augmentation
+# DIBBs Augmentation Lambda
 
-**General disclaimer** This repository was created for use by CDC programs to collaborate on public health related projects in support of the [CDC mission](https://www.cdc.gov/about/cdc/#cdc_about_cio_mission-our-mission). GitHub is not hosted by the CDC, but is a third party website used by CDC and its partners to share information and collaborate on software. CDC use of GitHub does not imply an endorsement of any one particular service, product, or enterprise.
```
Comment (JNygaard-Skylight) [inline packages/augmentation-lambda/README.md:3]: Does this disclaimer need to be in each package's readme?

File: packages/augmentation-lambda/README.md [hunk 12]
```diff
@@ -1,19 +1,62 @@
...
+AUGMENTED_EICR_PREFIX
+AUGMENTATION_METADATA_PREFIX
+```
+
+## Tests
+
+Run the package tests with:
+
+```bash
+uv run pytest packages/augmentation-lambda/tests
```
Comment (JNygaard-Skylight) [inline packages/augmentation-lambda/README.md]:
```suggestion
just test all packages/augmentation-lambda/tests
```

File: packages/augmentation-lambda/README.md (hunk 11 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md:3]: probably not. I can't picture a scenario where the lambdas aren't used in tandem, but that was my initial thought for including it everywhere.

File: packages/augmentation-lambda/README.md (hunk 1 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md:53]: updated!

File: packages/augmentation-lambda/README.md (hunk 11 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md:3]: disclaimer is in the main readme, so I'll remove for now and we can always add back if needed.

File: packages/augmentation-lambda/README.md (hunk 2 above)
Comment (robertandremitchell) [reply packages/augmentation-lambda/README.md]: updated!

File: packages/index-lambda/README.md (hunk 3 above)
Comment (robertandremitchell) [reply packages/index-lambda/README.md]: makes sense, removing

File: packages/lambda-handler/README.md (hunk 5 above)
Comment (robertandremitchell) [reply packages/lambda-handler/README.md:57]: makes sense, updating

File: packages/lambda-handler/README.md (hunk 4 above)
Comment (robertandremitchell) [reply packages/lambda-handler/README.md]: updated!

File: packages/index-lambda/README.md (hunk 10 above)
Comment (robertandremitchell) [reply packages/index-lambda/README.md:3]: weird copilot change. removed block anyway

File: packages/text-to-code-lambda/README.md (hunk 6 above)
Comment (robertandremitchell) [reply packages/text-to-code-lambda/README.md]: updated! lmk if that makes sense

**Review by nickclyde — APPROVED** [review]
Thanks for addressing Rob!!
